### PR TITLE
test: Cypress - Flaky Fix

### PR DESCRIPTION
**Pull request by Vijetha-Kaja**
## Description

**Fixed below flaky tests**
- Button_onClickAction_spec.js
- TemplatesPage_filtering_spec.ts
 
## Type of change

- Flaky test fix

## How Has This Been Tested?
- Cypress test runs

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js [hunk 1]
```diff
@@ -38,61 +38,61 @@ describe("Button Widget Functionality", function () {
...
+    entityExplorer.ExpandCollapseEntity("Container3");
+    entityExplorer.SelectEntityByName("Button1");
 
     // Adding the query in the onClickAction of the button widget.
     cy.executeDbQuery("Query1", "onClick");
     // Filling the messages for success/failure in the onClickAction of the button widget.
     cy.onClickActions("Success", "Error", "Execute a query", "Query1.run");
 
-    _.deployMode.DeployApp();
+    deployMode.DeployApp();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js:74]: Wait for Button to be visible in `DeployApp`.

File: app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js [hunk 2]
```diff
@@ -38,61 +38,61 @@ describe("Button Widget Functionality", function () {
...
-    cy.get(publishPage.buttonWidget).click();
+    agHelper.GetNClick(publishPage.buttonWidget);
     cy.get("body").then(($ele) => {
       if ($ele.find(widgetsPage.apiCallToast).length <= 0) {
-        cy.get(publishPage.buttonWidget).click();
+        agHelper.GetNClick(publishPage.buttonWidget);
       }
     });
-    cy.get(widgetsPage.apiCallToast).should("have.text", "Success");
+    agHelper.GetNAssertElementText(widgetsPage.apiCallToast, "Success", "contain.text");
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js:83]: Use `ValidateToast()`

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js:83]: Validate toast is using some other selector to validate toast message whereas button_onclick is using different selector

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: Goal here is to validate toast message right?

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: yes

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: Then locator used does not matter right?

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: no no, we are using predefined locator in `validateToastMessage` method to check for toast message, i remember while migrating specs to Ads this predefined locator didn't work for this case as the toast is having different locator here. That's the reason for not using `validateToastMessage` method

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: let me check once again

Comment (Vijetha-Kaja) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: It worked. Thanks!

Comment (Aishwarya-U-R) [reply app/client/cypress/e2e/Regression/ClientSide/Widgets/Button/Button_onClickAction_spec.js]: Why other test cases were not updated?! 🤔

**Pull request merged**